docs/howto_guides/Realtime-Best-Practices.md

```diff
@@ -0,0 +1,541 @@
+
+An **over-run** occurs if the executive does not finish running all of its scheduled jobs. This is a **bad-thing**. In this case, the executive will immediately start the next frame in an attempt to catch up.
+
+<a id=figure-Realtime-over-run></a>
+![Realtime Over Run](images/RealtimeOverRun.png)
+
+---
+
+<a id=itimers></a>
+### Itimers ( Being Nice to Other Processes On Your System )
+
+During real time under runs you may want to release the processor for other tasks to use instead of spinning waiting for the clock. Trick provides a realtime sleep timer based on itimers. You might think of it as a “snooze button”.
+
+To enable itimers call ```trick.itimer_enable()``` from the input file.
+
+With itimer_enabled, the simulation will sleep() after completing the jobs scheduled for the current frame. The itimer will then wake the sim 2ms before the realtime frame is to expire.  The executive will spin for the final 2ms. The 2ms spin is there because an itimer interval is not guaranteed to be perfectly precise.
+
+<a id=figure-Realtime-with_itimer></a>
+![Realtime with itimer](images/RealtimeWithItimer.png)
+
+[Ref: Itimer](https://nasa.github.io/trick/documentation/simulation_capabilities/Realtime-Timer)
+
+---
+
+<a id=frame-logging></a>
+### Frame-Logging ( Critical For Improving Sim Performance )
+
+The failure of a simulation to meet its scheduling requirements can have many causes. To aid in solving these problems, Trick can gather simulation performance data, called **frame-logging** by calling:
+
+```trick.frame_log_on()```
+
+in your sim's input file.
+
+Note that frame logging will add some overhead to a simulation as each job is timed and recorded.
+
+[Ref: Frame-Logging](https://nasa.github.io/trick/documentation/simulation_capabilities/Frame-Logging)
+
+<a id=frame-log-files></a>
+#### Frame Log Files
+Frame logging records the following data files in your sim’s RUN_ directory:
+
+* [```log_frame.trk```](#log-frame-trk)<br>
+* [```log_frame_trickjobs.trk```](#log-frame-trickjobs-trk)
+* [```log_frame_frame_userjobs_main.trk```](#log-frame-userjobs-main-trk)
+* [```log_timeline.csv```](#log-timeline-csv)
+* [```log_timeline_init.csv```](#log-timeline-csv)
+
+Note that **main** in this file name refers to the main thread.
+If child threads (for example: C1, C2, ...) have been specified in the sim then frame log files for those threads will also be created ( that is:```log_trick_frame_userjobs_C1.trk```, ```log_trick_frame_userjobs_C2.trk```, ```...```).
+
+---
+
+<a id=log-frame-trk></a>
+#### ```log_frame.trk```
+* Number of fields per record : 5
+
+|#  | Name | Type | Units | Description |
+|--:|:---- |:-----|:-----:|-------------|
+| 1| ```sys.exec.out.time```                  | double | seconds |Simulation Time |
+| 2| ```trick_real_time.rt_sync.frame_time``` | double | seconds | This badly named parameter expresses the amount of time that the scheduled jobs in this frame took to execute. See: [figure](#figure-realtime-under-run)|
```

Comment:
(sorry for commenting on a merged pull request, but I'm trying to use this page and am confused by) it seems that this `frame_time` can go above the frame period while the `frame_overrun` remains negative. See, for example, @keithvetter's comment in https://github.com/nasa/trick/issues/1609#issuecomment-1863380623

How is this possible? If the scheduled jobs took more than the alloted frame period, isn't that an overrun according to the linked figure above?

![image](https://github.com/nasa/trick/assets/34431661/f76a9cb8-c894-44e0-9bfe-922ff3add472)